src/components/NewAssignment.js

```diff
@@ -0,0 +1,147 @@
+
+    // Find the selected course object based on the course title
+    const selectedCourse = courses.find(
+      (courseObj) => courseObj.courseTitle === course
+    );
+
+    if (!selectedCourse) {
+      console.error("Selected course not found.");
+      return;
+    }
+
+    const assignmentData = {
+      assignmentName: name,
+      dueDate,
+      courseTitle: selectedCourse.courseTitle,
+      courseId: selectedCourse.courseId,
+    };
+
+    console.log("Assignment data:", assignmentData);
+
+    // Make the POST request
+    fetch(`${SERVER_URL}/course/${selectedCourse.courseId}/assignments`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-XSRF-TOKEN": token,
+      },
+      body: JSON.stringify(assignmentData),
+    })
+      .then((response) => {
+        if (response.ok) {
+          setSuccessMessage("Assignment added!");
+            // Clear the form after successful submission
+          setName("");
+          setDueDate("");
+          setCourse("");
+          return response.json();
+        } else {
+          setErrorMessage("Error encountered");
+          throw new Error("Error encountered");
+        }
+      })
+      .then((data) => {
+        console.log("Assignment created:", data);
+      })
+      .catch((error) => {
+        if (error.message !== "SyntaxError: Unexpected end of JSON input") {
+          console.error("Error encountered: ", error);
+        }
+      });
+  };
+
+  return (
+    <div>
+      <h2>New Assignment</h2>
+      <form onSubmit={handleSubmit}>
+        <div>
+          <label htmlFor="name">Name: </label>
+          <input
+            type="text"
+            id="name"
+            value={name}
+            onChange={(e) => setName(e.target.value)}
+          />
+        </div>
+        <div>
+          <br />
+          <label htmlFor="dueDate">Due Date: </label>
+          <input
+            type="text"
+            id="dueDate"
+            value={dueDate}
+            onChange={(e) => setDueDate(e.target.value)}
+            placeholder="yyyy-mm-dd"
+          />
+        </div>
+        <div>
+          <br />
+          <label htmlFor="course">Course:</label>
+          {courses.length > 0 ? (
+            courses.map((courseObj) => (
+              <div key={courseObj.courseId}>
+                <input
+                  type="radio"
+                  id={courseObj.courseId}
+                  name="course"
+                  value={courseObj.courseTitle}
+                  checked={course === courseObj.courseTitle}
+                  onChange={(e) => setCourse(e.target.value)}
+                />
+                <label htmlFor={courseObj.courseId}>
+                  {courseObj.courseTitle}
+                </label>
+              </div>
+            ))
+          ) : (
+            <div>No courses available</div>
+          )}
+        </div>
+        <br />
+        <button type="submit">Create Assignment</button>
```

Comment:
Not an error, more of a style consistency thing: you could use MUI components for the form here (like this https://mui.com/material-ui/react-text-field/)